### ISSUE TITLE
Banner image cover tile type

### ISSUE THREAD

**Issue by skleinfeldt** [jazkarta/jazkarta.com#19]
We need a cover tile for a banner image that bleeds to the edge of the page, per the homepage design. This may also be used on landing pages. It must be possible to add/edit the image to use. We do not need text overlaid on top of the image for now.

I need help understanding what to do to ensure the image aspect ratio will be right. Does it need to be pre-cropped? Can it be auto cropped? Do we need to include a cropping tool like on ISAW? Maybe for now it should be pre-cropped and another ticket created for adding a cropping tool later?

**Comment by davisagli** [comment]
This was implemented during the sprint in d8fea0b8b7bfa3f3eb45999b6520dccef2f59d89 but maybe it hasn't been deployed yet.

It's up to you to crop the image before uploading. You should upload a fairly large image since it will be full width on large monitors.

**Comment by davisagli** [comment]
I tried setting this up but editing the banner tile currently gives an error:

```
Traceback (innermost last):

Module ZPublisher.Publish, line 138, in publish
Module ZPublisher.mapply, line 77, in mapply
Module ZPublisher.Publish, line 48, in call_object
Module plone.z3cform.layout, line 66, in __call__
Module plone.z3cform.layout, line 60, in update
Module plone.app.tiles.browser.base, line 60, in render
Module z3c.form.form, line 161, in render
Module zope.browserpage.viewpagetemplatefile, line 51, in __call__
Module zope.pagetemplate.pagetemplate, line 132, in pt_render
Module five.pt.engine, line 93, in __call__
Module z3c.pt.pagetemplate, line 163, in render
Module collective.newrelic.patches.chameleon_patch, line 15, in monkeypatch
Module newrelic.api.function_trace, line 110, in literal_wrapper
Module chameleon.zpt.template, line 258, in render
Module chameleon.template, line 171, in render
Module 793888861a19eec5157ad9b5333c4808.py, line 91, in render
Module e2bca393f3c7c7c413f5e9494f3ed557.py, line 1801, in render_titlelessform
Module e2bca393f3c7c7c413f5e9494f3ed557.py, line 451, in render_fields
Module e2bca393f3c7c7c413f5e9494f3ed557.py, line 126, in render_widget_rendering
Module e2bca393f3c7c7c413f5e9494f3ed557.py, line 1069, in render_field
Module five.pt.expressions, line 161, in __call__
Module Products.Five.browser.metaconfigure, line 479, in __call__
Module zope.browserpage.viewpagetemplatefile, line 83, in __call__
Module zope.browserpage.viewpagetemplatefile, line 51, in __call__
Module zope.pagetemplate.pagetemplate, line 132, in pt_render
Module five.pt.engine, line 93, in __call__
Module z3c.pt.pagetemplate, line 163, in render
Module collective.newrelic.patches.chameleon_patch, line 15, in monkeypatch
Module newrelic.api.function_trace, line 110, in literal_wrapper
Module chameleon.zpt.template, line 258, in render
Module chameleon.template, line 191, in render
Module chameleon.template, line 171, in render
Module 310e3900239ce890864bcfbd4670f51a.py, line 591, in render
Module 310e3900239ce890864bcfbd4670f51a.py, line 466, in render_widget_wrapper
Module five.pt.expressions, line 161, in __call__
Module plone.app.z3cform.widget, line 510, in render
Module plone.app.z3cform.widget, line 477, in _base_args
Module plone.app.z3cform.widget, line 68, in _base_args
Module plone.app.z3cform.widget, line 466, in pattern
Module Products.CMFCore.utils, line 13, in check_getToolByName
Module Products.CMFCore.utils, line 120, in getToolByName
AttributeError: portal_membership
```

It appears to be because the form context does not have an acquisition chain to the root of the site, and the new rich text widget in Plone 5 (in plone.app.z3cform/widget.py) does not wrap the context like the old one in plone.app.textfield did here: https://github.com/plone/plone.app.textfield/blob/master/plone/app/textfield/widget.py#L47

We need to fix the Plone 5 widget to do the same thing.

**Comment by cguardia** [comment]
I confirmed that the regular text field form cover also has the same problem, so I went ahead and fixed it in a branch using the plone.app.textfield method. I added my fork to the sources.cfg, so running the buildout again will get that and fix the issue.

This is working now on the beta site.

**Comment by skleinfeldt** [comment]
If this is a fix for a Plone 5 bug we want to get the fix merged into P5 in addition to our short term site fix of using a fork. Can somebody do whatever is needed for that (submit a PR?) 

Will test as soon as I am cover and imagecrop competent.

**Comment by cguardia** [comment]
I can do the PR if @davisagli or @cewing take a quick look and think that's ok:

https://github.com/cguardia/plone.app.z3cform/commit/090b5ade2b61c6433c038b3fa4f06b03172e263c

**Comment by davisagli** [comment]
Thanks, @cguardia. That looks right.

**Comment by cguardia** [comment]
PR merged. Switched dev sources to main project's master, until the next release is out.

**Comment by skleinfeldt** [comment]
Maybe should be pinned?

**Comment by davisagli** [comment]
Whether we pin it or not we still have to remember to switch it back to the standard Plone package once the next Plone 5 release is ready.

**Comment by davisagli** [comment]
The banner tile is finally working. Try adding an image to it.

**Comment by skleinfeldt** [comment]
So nice!
